### PR TITLE
feat(updates.jenkins.io): add tools and credentials to `agent.trusted.ci.jenkins.io` for buckets and file share interactions

### PR DESCRIPTION
**Pull request by lemeurherve**
This PR adds the necessary tools and credentials needed to synchronize the R2 buckets and the Azure File Storage in https://github.com/jenkins-infra/update-center2/blob/master/site/publish.sh

Ref: https://github.com/jenkins-infra/helpdesk/issues/2649

### CODE REVIEW

File: dist/profile/manifests/buildagent.pp [hunk 1]
```diff
@@ -67,6 +72,46 @@
...
+      $architecture = 'arm64'
+    } else {
+      $architecture =  $facts['os']['architecture']
+    }
+    $azcopy_url = "https://azcopyvnext.azureedge.net/releases/release-10.21.0-20230928/azcopy_linux_${architecture}_10.21.0.tar.gz"
+    notify { "azcopy_url: ${azcopy_url}": }
+
+    exec { 'Install azcopy':
+      require => [Package['curl'], Package['tar'], Account[$user]],
+      command => "/usr/bin/mkdir -p /tmp/azcopy && /usr/bin/curl ${azcopy_url} | /usr/bin/tar -xz --strip-components=1 -C /tmp/azcopy && /usr/bin/cp /tmp/azcopy/azcopy /usr/local/bin/azcopy && /usr/bin/chmod +x /usr/local/bin/azcopy && /usr/bin/rm -rf /tmp/azcopy/",
```
Comment (dduportal) [inline dist/profile/manifests/buildagent.pp]:
```suggestion
      command => "/usr/bin/curl --silent --location ${azcopy_url} | /usr/bin/tar --extract --gzip --directory=/usr/local/bin/ --strip-components=1 --wildcards '*/azcopy'",
```

(nit)

Comment (lemeurherve) [reply dist/profile/manifests/buildagent.pp]: I tried something like that previously (extracting only the `azcopy` file) but didn't get it working, thanks!

Addressed in https://github.com/jenkins-infra/jenkins-infra/pull/3099/commits/2586af2f871a3f72edf18f2c00ed03055ff7fc90

File: dist/profile/manifests/buildagent.pp [hunk 2]
```diff
@@ -67,6 +73,60 @@
...
 
+    # There is no linux_aarch64 azcopy release, considering that aarch64 = amd64 so vagrant can run on Mac Silicon
+    $architecture = $facts['os']['architecture'] ? {
+      'aarch64' => 'arm64',
+      default   => $facts['os']['architecture'],
+    }
+    $azcopy_url = "https://azcopyvnext.azureedge.net/releases/release-10.21.0-20230928/azcopy_linux_${architecture}_10.21.0.tar.gz"
+
+    exec { 'Install azcopy':
+      require => [Package['curl'], Package['tar'], Account[$user]],
```
Comment (dduportal) [inline dist/profile/manifests/buildagent.pp]:
```suggestion
      require => [Package['curl'], Package['tar']],
```

Didn't realize earlier but the account creation is not required for this to be executed

Comment (lemeurherve) [reply dist/profile/manifests/buildagent.pp]: Right, it's from an earlier version in which I `chown`ed the binary.

**Review by dduportal — APPROVED** [review]

**Pull request merged**